AntistasiOfficial.Altis/Missions/SUP_CitySupply.sqf

```diff
@@ -0,0 +1,227 @@
+_endTime = dateToNumber _endTime;
+/*
+_FIAMarkers = mrkFIA + campsFIA;
+
+while {true} do {
+	sleep 0.1;
+	_posSupply = [_targetPosition,2000,random 360] call BIS_fnc_relPos;
+	_nFIAMarker = [_FIAMarkers,_posSupply] call BIS_fnc_nearestPosition;
+	if ((!surfaceIsWater _posSupply) && (_posSupply distance _posHQ < 4000) && (getMarkerPos _nFIAMarker distance _posSupply > 500)) exitWith {};
+};
+*/
+_houseType = "Land_i_Shed_Ind_F";
+_crateType = "Land_PaperBox_01_open_boxes_F";
+if(_type == "WATER") then
+{
+	_crateType = "Land_PaperBox_01_open_water_F";
+};
+if(_type == "FUEL") then
+{
+	_crateType = "CargoNet_01_barrels_F";
+};
+_allSheds = nearestObjects [_targetPosition, [_houseType], 2000, true];
+if (count _allSheds == 0) exitWith
+{
+	diag_log format ["Supply mission not created, could not find %1 around %2", _houseType, _targetPosition];
+};
+
+_spawnPosition = _allSheds selectRandom;
+_spawnPosition = _spawnPosition findEmptyPosition [5,50, _crateType];
+sleep 1;
+_crate = _crateType createVehicle _spawnPosition;
+_crate allowDamage false;
+[_crate] spawn {sleep 1; (_this select 0) allowDamage true;};
+_crate call jn_fnc_logistics_addAction;
+//Spawned the crate in
+
+/*
+_marker = createMarker [format ["REC%1", random 100], _posSupply];
+_marker setMarkerShape "ICON";
+
+_task = ["SUP",[side_blue,civilian],[[_tskDesc,_targetName,numberToDate [2035,_endTime] select 3,numberToDate [2035,_endTime] select 4],_tskTitle,_marker],_posSupply,"CREATED",5,true,true,"Heal"] call BIS_fnc_setTask;
+
+*/
+
+_allGroups = [];
+_allSoldiers = [];
+
+_groupType = [infSquad, side_green] call AS_fnc_pickGroup;
+_group = [_posSupply, side_green, _groupType] call BIS_Fnc_spawnGroup;
+_allGroups pushBack _group;
+
+{
+	_group = _x;
+	{
+		[_x] spawn genInit; //Or genInitBASES ?? Where is the difference between them?
+		_allSoldiers pushBack _x;
+	} forEach units _group;
+} forEach _allGroups; //Should be only one by now so ok
+
+diag_log "SUP_CitySupply successful created";
+
+// wait until the player loads the crate
+waitUntil {sleep 1; (dateToNumber date > _endTime) OR !(isNull attachedTo _crate)};
+
+
+//if timer ran out
+if (dateToNumber date > _endTime) exitWith
+{
+	diag_log "SUP_CitySupply ended without the player even touching the crate";
+};
+
+if("SUP" in misiones) then
+{
+	{
+		if (isPlayer _x) then{
+			["You already have a mission of that kind, complete this first"] remoteExec ["hint",_this select 2];
+		}
+	} forEach ([100,0,_crate,"BLUFORSpawn"] call distanceUnits);
+	waitUntil{sleep 1;!("SUP" in misiones) OR (dateToNumber date > _endTime)};
+}
+
+_targetMarker = createMarker [format ["REC%1", random 100], _targetBuilding];
+_targetMarker setMarkerShape "ICON";
+
+_task = ["SUP",[side_blue,civilian], [[_tskDesc_drop, _targetName, numberToDate [2035,_endTime] select 3, numberToDate [2035,_endTime] select 4],_tskTitle,_targetMarker],_targetBuilding,"CREATED",5,true,true,"Heal"] call BIS_fnc_setTask;
```

Comment:
@StefArma I know you dont wanted a task, I thought about something, but if you dont like it, we can get rid of it.

You get a task telling you where to bring the supplies once you loaded them. That causes the restriction of only one delivery at the time, so I changed the spawn so it will only spawn one per tick

Reply:
the point is: player should be able to decide where to bring supplies
otherwise we just moved the starting point from HQ to this large building and added a few cosmethic variables

Reply:
Ok this will make this alot more complex, but I see the point.
So plan is:
-You can find this crates around on altis (how many per tick?)
-You can bring it to a random city which need the supplies
-Before you finished you can abort and decide to bring the crate somewhere else